app/Community/Controllers/ForumTopicController.php

```diff
@@ -91,4 +98,110 @@ public function destroy(ForumTopic $topic): void
     {
         $this->authorize('delete', $topic);
     }
+
+    private function getTotalRecentForumTopics(int $permissions = Permissions::Unregistered): int
+    {
+        return ForumTopic::query()
+            ->distinct('ForumTopic.ID')
+            ->leftJoin('Forum', 'Forum.ID', '=', 'ForumTopic.ForumID')
+            ->leftJoin('ForumTopicComment', function ($join) {
+                $join->on('ForumTopicComment.ForumTopicID', '=', 'ForumTopic.ID')
+                    ->where('ForumTopicComment.Authorised', '=', 1);
+            })
+            ->where('ForumTopic.RequiredPermissions', '<=', $permissions)
+            ->whereNull('ForumTopic.deleted_at')
+            ->count('ForumTopic.ID');
+    }
+
+    public function recentlyActive(Request $request): InertiaResponse
+    {
+        $offset = $request->input('page', 1) - 1;
+        $count = 25;
+
+        /** @var User $user */
+        $user = auth()->user();
+        $permissions = Permissions::Unregistered;
+        if ($user) {
+            $permissions = (int) $user->getAttribute('Permissions');
+        }
+
+        $topics = $this->getRecentForumTopics(
+            page: (int) $request->input('page', 1),
+            permissions: $permissions,
+        );
+
+        $transformedTopics = array_map(
+            fn ($topic) => ForumTopicData::fromRecentlyActiveTopic($topic)->include(
+                'commentCount24h',
+                'oldestComment24hId',
+                'commentCount7d',
+                'oldestComment7dId',
+            ),
+            $topics
+        );
+
+        $paginator = new LengthAwarePaginator(
+            items: $transformedTopics,
+            total: $this->getTotalRecentForumTopics($permissions),
+            perPage: $count,
+            currentPage: $offset + 1,
+            options: [
+                'path' => $request->url(),
+                'query' => $request->query(),
+            ],
+        );
+
+        $paginatedTopics = PaginatedData::fromLengthAwarePaginator($paginator);
+
+        return Inertia::render('forums/recent-posts', [
+            'paginatedTopics' => $paginatedTopics,
+        ]);
+    }
+
+    private function getRecentForumTopics(int $page = 1, int $permissions = Permissions::Unregistered): array
+    {
+        $count = 25;
+        $offset = ($page - 1) * $count;
+
+        $query = "
+            SELECT ft.ID as ForumTopicID, ft.Title as ForumTopicTitle,
+                   f.ID as ForumID, f.Title as ForumTitle,
+                   lc.CommentID, lftc.DateCreated as PostedAt, lftc.author_id,
+                   ua.User AS Author, ua.display_name AS AuthorDisplayName,
+                   LEFT(lftc.Payload, 90) AS ShortMsg,
+                   LENGTH(lftc.Payload) > 90 AS IsTruncated,
```

Comment:
This limit seems significantly lower than the previous limit.

master:
![image](https://github.com/user-attachments/assets/d83e1aa4-8beb-43f3-a985-c90b418b6302)

this branch:
![image](https://github.com/user-attachments/assets/8c200fb6-430c-4a85-8366-bde7cae9fe94)

also the "..." appears to be missing.

Reply:
Fixed in latest -- when I migrated the query to the controller I didn't notice we were overriding the default length of 90 with a different value.

![Screenshot 2024-08-17 at 7 43 55 AM](https://github.com/user-attachments/assets/39e87cb7-6ef7-4d7a-a72b-9e8e96c6a9fa)